pkg/api/v1alpha1/cluster_types.go

```diff
@@ -42,6 +43,13 @@ type ProxyConfiguration struct {
 	NoProxy    []string `json:"noProxy,omitempty"`
 }
 
+type RegistryMirrorConfiguration struct {
+	// Endpoint defines the registry mirror to use for pulling image
+	Endpoint string `json:"endpoint,omitempty"`
+	// CACertSecretName defines the kubernetes secret name that stores the registry mirror CA certificate
+	CACertSecretName string `json:"caCertSecretName,omitempty"`
```

Comment:
Let's have a ref field and then under that reference the secretName. This would only get populated after the cluster is created if the cert is passed through the env var right?